configure.ac

```diff
@@ -117,4 +114,29 @@ AS_IF([test "x$with_docs_only" = xyes],
     ]) # Docs only
 
 
+AC_ARG_ENABLE([debug],
+              AS_HELP_STRING([--enable-debug], [Enable extra debugging code (default is NO).]),
+              [], [enable_debug=no])
+
+if test $enable_debug = yes; then
+	AC_DEFINE([ENABLE_DEBUG], [1], [Enable debugging code])
+	CFLAGS="$CFLAGS -O0 -g3"
+	AC_DEFINE([UCS_MAX_LOG_LEVEL], [UCS_LOG_LEVEL_TRACE_POLL], [Highest log level])
+else
+	AC_DEFINE([UCS_MAX_LOG_LEVEL], [UCS_LOG_LEVEL_INFO], [Highest log level])
+	CFLAGS="$CFLAGS -O3 -DNDEBUG"
+fi
+
+m4_include([m4/ucx.m4])
+CHECK_UCX
+AC_MSG_RESULT([UCX support: $ucx_happy])
```

Comment:
since ucx/ucs is a requirement then configure script should fail if ucx wasn't found

Reply:
agreed. fixed. Added minimal required version check: UCX 1.9 - before that UCS does not have required config_read functionality.